internal/cache/persistent_lru_cache.go

```diff
@@ -231,6 +231,7 @@ func (c *PersistentCache) Put(ctx context.Context, key string, data gather.Bytes
 		// MUST NOT go over the specified limit for the cache space to avoid
 		// snapshots/restores from getting affected by the cache's storage use.
 		if c.isCacheFullLocked() {
+			log(ctx).Warnf("Cache is full, unable to add %v into cache.", key)
```

Comment:
we should add some kind of rate limiter here, for example by logging this message no more than once/minute, otherwise it will spam the logs.

Reply:
alternatively make that configurable via a flag or environment variable.

Reply:
I added a condition to limit to one warning max per minute, is the implementation ok ?